tests/deployment/test_client_server.py

```diff
@@ -97,172 +99,192 @@ def test_client_server_sklearn(
         x_train,
         configuration=default_configuration,
-        n_bits=n_bits,
-        global_p_error=1 / 100_000,
+        n_bits=2,
     )
 
+    # Check that client and server files are properly generated
+    check_client_server_files(quantized_numpy_module)
+
     # Check that the FHE execution is correct.
-    # With a global_p_error of 1/100_000 we only allow one run.
     check_is_good_execution_for_cml_vs_circuit(
         x_test, quantized_numpy_module, simulate=False, n_allowed_runs=1
     )
 
-    client_server_simulation(x_train, x_test, quantized_numpy_module, default_configuration)
-
-
-def client_server_simulation(x_train, x_test, model, default_configuration):
-    """Simulate the client server interaction."""
-    # Model has been trained and compiled on the server.
-    # Now we use the fhe api to go into production.
-
-    # Set up the fake network
-    network = OnDiskNetwork()
-
-    # Instantiate the dev client and server FHEModel client server API
-    fhemodel_dev = FHEModelDev(path_dir=network.dev_dir.name, model=model)
-    fhemodel_dev.save()
-
-    # Check that the processing json file is in the client.zip file
-    with zipfile.ZipFile(Path(network.dev_dir.name) / "client.zip") as client_zip:
-        with client_zip.open("serialized_processing.json", "r") as file:
-            assert isinstance(json.load(file), dict)
-
-    # Send necessary files to server and client
-    network.dev_send_clientspecs_and_modelspecs_to_client()
-    network.dev_send_model_to_server()
-
-    # Make sure the save fails now that the folder is populated
-    err_msg = (
-        f"path_dir: {network.dev_dir.name} is not empty."
-        "Please delete it before saving a new model."
-    )
-    with pytest.raises(Exception, match=err_msg):
-        fhemodel_dev.save()
-
-    fhemodel_client = FHEModelClient(
-        path_dir=network.client_dir.name,
-        key_dir=default_configuration.insecure_key_cache_location,
+    check_client_server_execution(
+        x_test, quantized_numpy_module, key_dir, check_array_equal, check_float_array_equal
     )
-    fhemodel_client.load()
 
-    # Grab the model and save it again
-    # No user is expected to load a FHEModelDev instance from a FHEModelClient's model. This is
-    # only made a testing for making sure the model has the expected attributes
-    client_model = fhemodel_client.model
-    client_model.fhe_circuit = model.fhe_circuit
 
-    # pylint: disable-next=protected-access
-    client_model._is_compiled = True
-
-    network.cleanup()
+def check_client_server_files(model):
+    """Test the client server interface API generates the expected file.
 
+    This test expects that the given model has been trained and compiled in development.
+    """
     # Create a new network
-    network = OnDiskNetwork()
+    disk_network = OnDiskNetwork()
 
     # And try to save it again
-    fhemodel_dev_ = FHEModelDev(path_dir=network.dev_dir.name, model=client_model)
```

Comment:
I basically spitted the initial test in two tests : 
- client/server file generation test
- predict result tests

It was not clear to me why we were loading two `FHEModelDev` objects using `client_model` in the first place. Comments were saying that this was for making sure the client's model had all necessary attributes, but I believe that checking the file generation + computing the predictions is more than enough (especially since it required this test to manually transfer attributes between instances, something that should never happen)